src/Domain/ElementMap.cpp

```diff
@@ -53,6 +55,52 @@ ElementMap<Dim, TargetFrame>::ElementMap(
       jacobian_{map_slope_},
       inverse_jacobian_{map_inverse_slope_} {}
 
+// We could refactor the ElementMap class to use a `Composition` internally also
+// for the element-to-block-logical map, so the whole map is just one
+// composition.
+template <size_t Dim, typename TargetFrame>
+ElementMap<Dim, TargetFrame>::ElementMap(ElementId<Dim> element_id,
+                                         const Block<Dim>& block)
+    : ElementMap(
+          std::move(element_id),
+          [&block]() -> std::unique_ptr<domain::CoordinateMapBase<
+                         Frame::BlockLogical, TargetFrame, Dim>> {
+            if constexpr (std::is_same_v<TargetFrame, Frame::Inertial>) {
+              if (block.is_time_dependent()) {
+                if (block.has_distorted_frame()) {
```

Comment:
Do you actually need this case? Shouldn't the else branch work even if there's a distorted frame? I'm fairly confident the evolution code at least requires grid->inertial to always be a valid map, independent of whether or not there's a distorted frame.

Reply:
You're right 👍 removed the unnecessary case.